### ISSUE TITLE
Create documentation for lldb/gdb -> aglib function mapping

### ISSUE THREAD

**Issue by fidgetingbits** [pwndbg/pwndbg#2522]
As suggested in this [PR](https://github.com/pwndbg/pwndbg/pull/2515#issuecomment-2470625879), if you are porting code that uses other ldb/gdb python module, it can be difficult to quickly figure out which pwndbg aglib function should be the appropriate replacement.

For instance `gdb.info.address("foo")` currently becomes `pwndbg.dbg.selected_inferior().symbol_address_from_name("foo")` (though that will change after #2521). It would be useful for a markdown table or some other documentation to exist to be able to easily lookup what functions map to what.

**Comment by mbrla0** [comment]
Yeah, we really a need something that explains the internal structure of Pwndbg as it is post-LLDB port. I realize now I should've probably typed something up sooner. But, in my defense, it's only now that people have actually started contributing to the debugger-agnostic bits. I'll come up with something that can at least help explain the new interfaces soon.

**Comment by mbrla0** [comment]
I've submitted a tentative change that should clear some of these things up. Though I had a hard time coming up with examples of things to show the `aglib` mappings. Can you give me a few more examples of things you'd like to see?

**Comment by disconnect3d** [comment]
We have a lot of notes on aglib etc in https://pwndbg.re/stable/contributing/dev-notes/ 

Closing
